ts/react-responsive-components.tsx

```diff
@@ -130,6 +130,13 @@ export interface ResponsiveProps extends MixedInHoCProps {
 	currentBreakpoint?: Breakpoint
 	maxSize?: string
 	minSize?: string
+	// Children has been removed from React FC types, but we heavily rely on that with web responsive components
+	// https://github.com/DefinitelyTyped/DefinitelyTyped/pull/56210
+	// We refine children type as the following because in `Responsive` component's option 2, we are able
+	// to pass a function as the children of the Responsive component.
+	// Usually responsiveKey is a string type, but in your own repo, you may define a set of string as enum for its type.
+	// Hence we use any here to cover any customised types.
+	children?: React.ReactNode | undefined | ((responsiveKey: any) => JSX.Element)
```

Comment:
Did we use `any` all throughout for `responsiveKey`?

Reply:
We definitely identified `responsiveKey` as `string` all throughout this library. If we use `string` here does it not widen automatically when we pass `ResponsiveKey` (type union of strings) to it?

Reply:
It should be string normally, what we did is we defined a type for it in our repo, like this:
```
export type ResponsiveKey = "narrowest" | "extraNarrow" | "narrow" | "medium" | "wide" | "widest"
```

But string should still work, let me change it to `string` instead of `any`.

Reply:
Updated.